test/partitioning.jl

```diff
@@ -85,6 +85,16 @@
     p = partition(grid, BlockPartition(T(5),T(2)))
     @test length(p) == 12
     @test Set(nelements.(p)) == Set([5,10])
+
+    grid = CartesianGrid{T}(100,100,100)
+
+    p = partition(grid, BlockPartition(T(1.), T(1.), T(1.), neighbors = true))
+    @test length(p) == 1000000
+    @test Set(nelements.(p)) == Set(1)
+
+    p = partition(grid, BlockPartition(T(5.), T(5.), T(5.), neighbors = false))
+    @test length(p) == 8000
+    @test Set(nelements.(p)) == Set(125)
```

Comment:
Can you share the times for these two tests specifically? We also don't want to increase the time that it takes to run the test suite considerably so I would just increase the size of the blocks to something else if it is taking more than 2s for example. Also you need to check what this PR is actually doing, which is providing a new option that computes or not the metadata. So you need to add tests to check `metadata(p)` in both cases.

Reply:
Working on it now!

Reply:
When neighbors is true, blocksize of 1 is making the function take 30+ seconds. However, if I set neighbors to false for blocksize (1,1,1) and true for size (5,5,5), it finishes in 7 seconds. Would this be alright?

Reply:
Adding metadata test now

Reply:
Looking into it now!

Reply:
```julia
function testing123()
    grid = CartesianGrid(50,50,50)
    p = partition(grid, BlockPartition(1., 1., 1., neighbors = false))
    println(length(p) == 1000000)
    println(Set(nelements.(p)) == Set(1))

    p = partition(grid, BlockPartition(5., 5., 5., neighbors = true))
    println(length(p) == 8000)
    println(Set(nelements.(p)) == Set(125))
end
```
Takes a total of 0.836927 seconds to run. Should I update this?

Reply:
When I check for metadata, should I go for a complete value to value check, or should I only check whether the Dict is empty or not? I am having a bit of trouble reading the values since the matrix is quite large, which is leading to a lot of metadata calculation which I am not able to read.

Reply:
Okay :+1:

Reply:
Regarding the metadata test, you should see that the neighbors key will contain a vector with the same length of the number of blocks returned. Each entry will contain the neighboring blocks. For example, the block at index (1,1) has neighbors [(2,1),(1,2)]. You can just check the following:

```julia
p = partition(...)
m = metadata(p)
n = m[:neighbors]
@test length(n) == length(p)
@test all(0 .< length.(n) .<= 27)
```

Reply:
So I am checking that the vector `neighbors` has the same length of the number of blocks returned and also checking that each block returned has at most 3*3*3 = 27 blocks nearby.

Reply:
@juliohm I have added the metadata test for the 3D (50,50,50) grid itself with a blocksize of (5,5,5). Do let me know if the code looks too messy because of the size of the dict, I will try using something else instead in that case.

Reply:
Ohhhh Okay, this makes a lot more sense than what I was doing earlier. Thank you!

Reply:
I have added the changes.

Reply:
I have added the changes! Please do let me know if I should add some more tests.